common/src/main/java/org/apache/comet/vector/CometPlainVector.java

```diff
@@ -153,11 +153,7 @@ public CDataDictionaryProvider getDictionaryProvider() {
 
   @Override
   public boolean isNullAt(int rowId) {
-    if (this.valueBufferAddress == -1) {
-      return true;
-    } else {
-      return super.isNullAt(rowId);
-    }
+    return this.valueBufferAddress == -1 || super.isNullAt(rowId);
```

Comment:
I would expect this to have identical performance to the original (i.e. this should result in identical byte code).

Reply:
### Before
```
  public boolean isNullAt(int);
    Code:
       0: aload_0
       1: getfield      #45                 // Field valueBufferAddress:J
       4: ldc2_w        #42                 // long -1l
       7: lcmp
       8: ifne          13
      11: iconst_1
      12: ireturn
      13: aload_0
      14: iload_1
      15: invokespecial #195                // Method org/apache/comet/vector/CometDecodedVector.isNullAt:(I)Z
      18: ireturn
```

### After
```
  public boolean isNullAt(int);
    Code:
       0: aload_0
       1: getfield      #45                 // Field valueBufferAddress:J
       4: ldc2_w        #42                 // long -1l
       7: lcmp
       8: ifeq          19
      11: aload_0
      12: iload_1
      13: invokespecial #195                // Method org/apache/comet/vector/CometDecodedVector.isNullAt:(I)Z
      16: ifeq          23
      19: iconst_1
      20: goto          24
      23: iconst_0
      24: ireturn
```

So basically jump order is different and seems that is causing the difference at least on my machine...

Reply:
interesting find